client/components/select-vertical/suggestion-search.tsx

```diff
@@ -115,13 +115,15 @@ const SelectVerticalSuggestionSearch: FC< Props > = ( {
 		name,
 		label,
 		value,
+		has_vertical_images,
```

Comment:
I think there is a bigger root cause here, which could have caused the confusion.

This `handleSuggestionsSelect` function is passed as `suggest` prop here:

https://github.com/Automattic/wp-calypso/blob/88c50830e252207c065a1353546563d637648a01/packages/components/src/suggestions/index.tsx#L29

but the argument types are definitely not compatible. (E.g. `Suggestion` does not have `has_vertical_images` prop, and where is the `index` argument?)

Could we investigate more and fix this type mismatch?

Reply:
Since the `Suggestion` component is a generic component, I'm hesitant about extending its type to fit our needs. Which is why I handle the "transformation" on our `SelectVertical`. Unless I missed something, and there's another approach we can try?

Reply:
I see, sorry I just realized that `Suggestions` is an existing generic component, and the `Suggestion` type is not exported :(

Hmm. What if we don't destructure the props? This way, every field from `Vertical` will be passed, even if we add more in the future. Something like this:

```
const handleSuggestionsSelect = ( suggestion: {
    name: string;
    label: string;
    value?: string;
} ) => {
    hideSuggestions();
    onSelect?.( suggestion as Vertical );
};
```